### DISCUSSION TITLE
Filament runout sensor

### DISCUSSION THREAD

**Discussion by ClearAirTurbulence3D** [bassamanator/everything-sovol-sv06#1] [General]
I don't feel the need for a filament sensor - it's just more mass to sling around on the hotend. That said, the SV06 seems ready for one. The J7 pinouts (VGS - V (VDD= 3.3V or 5V) G (ground) S (signal) match the pinouts on (for example) the BigTreeTech smart-filament-detection-module: VSG

The filament sensor is commented out in the SV06 config.h file (line 1438) but the sensor pin is defined in pins_SOVOL_V131.h:

// Filament Runout Sensor
//
#ifndef FIL_RUNOUT_PIN
  #define FIL_RUNOUT_PIN                    PA4   // "Pulled-high"
#endif

**Comment by bassamanator** [comment]
Thanks for this. 😀

I have this info listed (but commented out) in my OSS Klipper config, however, I'm not sure how I would go about displaying it here exactly. I'll see if I can think of something. Please create a pull request about this if you already have some ideas. We just want to make sure things are very clear to whomever looks this up in the future.

**Comment by bassamanator** [comment]
I actually got a filament runout sensor working. The sensor is connected to the hotend PCB.

Once I have a complete solution, I'll be adding relevant details here, as well as updating my klipper config.

😁

**Comment by ndrwstn** [comment]
Any thoughts on the BTT SFS v2? It has both a runout and motion sensor, with a four pin connector. Obviously won’t directly plug up into the tool head three-pin, but I’d like to mount it away from the tool head anyway. Looking at the motherboard though, it doesn’t seem like there are my free pins though?

**Comment by ClearAirTurbulence3D** [comment]
It looks interesting. I used the previous version on an older printer. Nice on paper but very heavy and no easy way of mounting it (unless you printed a back shell for it). At least this one has mounting holes.If there’s a spare pin for the motion detection on the SV06 motherboard, it should work. Otherwise it’s just a break sensor, but more gentle on the filament.FrankOn Jul 30, 2024, at 2:43 PM, ndrwstn ***@***.***> wrote:﻿
Any thoughts on the BTT SFS v2? It has both a runout and motion sensor, with a four pin connector. Obviously won’t directly plug up into the tool head three-pin, but I’d like to mount it away from the tool head anyway. Looking at the motherboard though, it doesn’t seem like there are my free pins though?

**Comment by ndrwstn** [comment]
That's what I'm trying to find out--if the motherboard has a spare pin (without tearing it apart to investigate myself, as I try to keep my printer running). Looking at the various pictures online, it does not appear to have one.

I haven't figured out if I want to mount it on the toolhead to on the frame above. The toolhead is convenient, as it will allow me to run ptfe tube from my spool directly (as I'm not using the stock spoolholder), but adds a decent bit of weight. The frame adds some complications too, mostly in just ensuring a smooth path from spool to the frame and down to the toolhead without binding.

**Comment by ClearAirTurbulence3D** [comment]
The SV06 motherboard is basically the same as the Creality 3D Ender-3 V2 motherboard; there don't seem to be any extra pins left over

**Comment by Dealman** [comment]
Might I hijack this to ask whether the voltage output for this header is 3.3V or 5V? Seen [a guide](https://sv06.blakadder.com/Upgrades/nozzle-led/) where they mount a LED strip in order to light up the tool head, connecting it to the filament sensor header.

Figured it's a neat upgrade since I have no need for a runout sensor.

**Reply by bassamanator** [reply]
Not sure if I have it listed there, but please check my other repo: https://github.com/bassamanator/everything-sovol-sv06
